src/change_dump_date.py

```diff
@@ -1,47 +1,171 @@
-#!/usr/bin/env python
-# Change the initial and valid date of a UM dump file
+import os
+import mule
+import argparse
 
-# Change both the file header and the date header of each record.
-# The latter may not be strictly necessary but it makes looking at file with
-# xconv less confusing.
+def year_value(value):
+    """
+    Ensure the year is a positive integer.
+    """
+    ivalue = int(value)
+    if ivalue < 1:
+        raise argparse.ArgumentTypeError(f"Invalid year: {value}. Must be a positive integer.")
+    return ivalue
 
-import getopt, sys
-import umfile
-from um_fileheaders import *
+def month_value(value):
+    """
+    Ensure the month is between 1 and 12 (inclusive).
+    """
+    ivalue = int(value)
+    if ivalue < 1:
+        raise argparse.ArgumentTypeError(f"Invalid month: {value}. Must be a positive integer.")
+    if ivalue > 12:
+        raise argparse.ArgumentTypeError(f"Invalid month: {value}. Must be between 1 and 12.")
+    return ivalue
 
-ifile = sys.argv[1]
+def day_value(value):
+    """
+    Ensure the day is between 1 and 31 (inclusive).
+    """
+    ivalue = int(value)
+    if ivalue < 1:
+        raise argparse.ArgumentTypeError(f"Invalid day: {value}. Must be a positive integer.")
+    if ivalue > 31:
+        raise argparse.ArgumentTypeError(f"Invalid day: {value}. Must be between 1 and 31.")
+    return ivalue
 
-f = umfile.UMFile(ifile, "r+")
 
-print("Initial Time", f.fixhd[FH_DTYear], f.fixhd[FH_DTMonth],
-      f.fixhd[FH_DTDay], f.fixhd[FH_DTHour], f.fixhd[FH_DTMinute],
-      f.fixhd[FH_DTSecond])
+def parse_args():
+    """
+    Parse the command line arguments.
 
-print("Valid Time", f.fixhd[FH_VTYear], f.fixhd[FH_VTMonth],
-      f.fixhd[FH_VTDay], f.fixhd[FH_VTHour], f.fixhd[FH_VTMinute],
-      f.fixhd[FH_VTSecond])
+    Parameters
+    ----------
+    None
 
-s = input('Enter year month day\n')
-s = s.split()
-year = int(s[0])
-month = int(s[1])
-day = int(s[2])
+    Returns
+    ----------
+    args_parsed : argparse.Namespace
+        Argparse namespace containing the parsed command line arguments.
+    """
+    parser = argparse.ArgumentParser(description="Modify UM dump file timestamps.")
+    parser.add_argument('ifile', metavar="INPUT_PATH", help='Path to the input file.')
+    parser.add_argument('-y', '--year', type=year_value, help='New year value.', required=True)
+    parser.add_argument('-m', '--month', type=month_value, help='New month value (1-12).', required=True)
+    parser.add_argument('-d', '--day', type=day_value, help='New day value (1-31).', required=True)
```

Comment:
```suggestion
    parser.add_argument('-d', '--day', type=day_value, help='New day value (1-31).')
```
I don't think it should be required. If I don't provide any day it should keep the current one in the file.